tests/e2e/specs/shopper/mini-cart.test.js

```diff
@@ -17,8 +17,8 @@ const options = getDefaultOptions();
 const clickMiniCartButton = async () => {
 	await page.hover( '.wc-block-mini-cart__button' );
 
-	await expect( page ).not.toMatchElement(
-		'.wc-block-mini-cart__drawer.is-loading'
+	await page.waitForFunction(
+		'document.querySelector(".wc-block-mini-cart__drawer.is-loading") === null'
 	);
```

Comment:
we can use 
```
page.waitForSelector( '.wc-block-mini-cart__drawer.is-loading', { hidden: true } )
```
`hidden: true` inverts selector from waiting for presence to waiting for removal/non-presence - what do you think?

Reply:
I'm a bit worried about `hidden` at first, as the docs says

> hidden <[boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#Boolean_type)> wait for element to not be found in the DOM or to be hidden, i.e. have display: none or visibility: hidden CSS properties. Defaults to false.

Because the drawer is hidden by default, I thought I can return immediately, which leads to clicking the Mini Cart button while it's still loading (which causes the flaky tests we're trying to solve).

But testing your suggestion locally several times, looks like it works as expected. Committing yours because I also prefer the declarative approach. Thanks!

Reply:
Cheers!